roles/kubernetes-apps/ansible/tasks/coredns.yml

```
@@ -22,6 +23,7 @@
     - dns_mode in ['coredns', 'coredns_dual']
     - inventory_hostname == groups['kube_control_plane'][0]
     - enable_dns_autoscaler or item.name != 'dns-autoscaler'
+    - item.condition | default(True) | bool
```

```suggestion
    - item.condition | default(True)
```
You don't need the bool filter now that you dropped the jinja templating ; unless I'm mistaken it should work without it.